docs/concepts/components.md

```diff
@@ -86,3 +83,19 @@ Any pod may mount an Ondat virtual volume from any node that is also
 running Ondat, regardless of whether the pod and volume are
 collocated on the same node. Therefore, applications may be started or
 restarted on any node and access volumes transparently.
+
+## Ondat Upgrade Guard
+
+The upgrade guard is a key component of the rolling upgrade feature. It blocks certain nodes from being upgraded or drained thus avoiding data loss in the cluster.
+
+The upgrade guard will detect if a volume is reconciling (for example, one that does not have enough synced replicas), at which point a node manager pod on the same node as the reconciling volume's master and replicas become unready. Ondat uses a PodDisruptionBudget (PDB) to stop more than 1 node manager pod being unavailable at any point in time. This prevents the rolling upgrade from continuing until the PDB is satisfied and all volumes have fully reconciled
+
+If the PDB is set to 1 and a Control Plane volume on a node is not ready for a long period of time, this will stop the upgrade process. The `api-managercomponent` will be able to dynamically set the PDB value if it can determine the health of the volume. If the `api-managercomponent` knows that a volume will not be ready, it can increase the PDB `maxUnavailable` value, allowing the upgrade to continue. The upgrade guard container will log when it is available to upgrade, it will also log the reason if upgrade is not possible.
+
+>⚠️ The upgrade guard container only monitors volumes that host a deployment on its node (for example, it doesn’t care if a volume is unhealthy if the node it's running on hosts none of the volumes primary and replicas)
+
+>⚠️ There is some latency between a volume becoming unhealthy and the upgrade guard noticing, due to the polling nature of both the `api-managercomponent` volume sync Kubernetes readiness endpoints)
+
```

Comment:
Limitations on different providers is missing. Could you please add the following to the docs?

```
> Limitations:
GKE, AKS takes care of the pod disruption budget for one hour. After this time period they drain the node, which should destroy the volume
EKS takes care on PDB for 50 mins, after this period upgrade fails unless it was forced
```

Reply:
My bad i have found it at the end :)